lib/spack/spack/util/executable.py

```diff
@@ -326,7 +328,7 @@ def which(*args, **kwargs):
         Executable: The first executable that is found in the path
     """
     exe = which_string(*args, **kwargs)
-    return Executable(shlex.quote(exe)) if exe else None
+    return Executable(exe) if exe else None
```

Comment:
I assume this can undo the `shlex.quote` from https://github.com/spack/spack/commit/4ff8a6a9b703002db9f1c5a949a4b00c4978363d because we no longer `shlex.split`.

Reply:
yes that's right